revise lua script to use zig tool chain for stride

jazzay opened the pull request:
These changes are required to support building NativePath and Celt for newer platforms like osx-arm64.

This will have no impact on Stride itself until I push the changes to deps, which would be into the Stride repo.

jazzay committed:
revise lua script to use zig tool chain for stride

jazzay committed:
fix issue with linker source libs

manio143 commented on tools/np-build-stride.lua:
```diff
@@ -0,0 +1,149 @@
+	
+	-- lfs.mkdir("macOS")
+	
+	if is_verbose == true then
```
Is CMD printing already covered in helper methods?

jazzay replied:
This was in the original impl, and looks to be unneeded, so I've removed.

manio143 commented on tools/np-build-driver-stride.lua:
```diff
@@ -0,0 +1,113 @@
+platform = "windows"
+outputName = "lib"
+
+function table.contains(table, element)
```
Can you move generic helpers like this, the string ones above, dirtree and Slash choosing to another file and only keep custom build related logic here?

jazzay replied:
I could yes, but I really just wanted to leave the original impl alone, and so I created these complete duplicates so I could apply the cleanup to just what Stride actually needs. Keeping the old intact is useful if / when we need to review it for other runtime support. And I think mobile will still need to use some of that old approach. But not sure yet.

I've already spent way too much time on this stuff, and don't want to risk breaking anything more, so if you guys are ok with it I'd prefer not to refactor this too much, for minimal value.

jazzay replied:
Ok I've just eliminated the stride specific driver file, and moved the few small changes into the common shared one now.

manio143 commented on tools/np-build-driver-stride.lua:
```diff
@@ -0,0 +1,113 @@
+    end
+end
+
+for filename, attr in dirtree(directory) do
```
Can you add a short comment above this loop what it's doing? It's clear, but it's easier to read through the file by just glancing at the comment

jazzay replied:
Will add a comment.

manio143 commented on tools/np-build-driver-stride.lua:
```diff
@@ -0,0 +1,113 @@
+for i,v in ipairs(arg) do 
+    if v == "debug" then
+        debug = true
+    elseif string.starts(v, "-I") then
```
Could use a note what does `-I` and `-D` mean here.

jazzay replied:
comments added

manio143 reviewed (approved):
Overall I think it looks good. Left some small comments.
Not sure how this Lua is part of the build to be honest, so an example on how to invoke it would be nice.

jazzay commented:
> Overall I think it looks good. Left some small comments. Not sure how this Lua is part of the build to be honest, so an example on how to invoke it would be nice.

This is old build tooling that predates the formation of the OSS Xenko. You can see where it is used to build NativePath inside deps/NativePath/build.bat. I have also added a new build.bat for the Celt dependency that uses this as well, since we need an update to that dep to support the new platform targets. I will be putting up a separate PR in the main Stride repo that illustrates changes related to this soon. Sadly this is in a separate repo.

jazzay committed:
address some PR feedback

jazzay commented:
> > Overall I think it looks good. Left some small comments. Not sure how this Lua is part of the build to be honest, so an example on how to invoke it would be nice.
> 
> This is old build tooling that predates the formation of the OSS Xenko. You can see where it is used to build NativePath inside deps/NativePath/build.bat. I have also added a new build.bat for the Celt dependency that uses this as well, since we need an update to that dep to support the new platform targets. I will be putting up a separate PR in the main Stride repo that illustrates changes related to this soon. Sadly this is in a separate repo.

Actually it was Celt dep that was using this already, I've added a new build.bat for NativePath that's incoming.

manio143 reviewed (approved):
I'm happy with it but would have @xen2 look it over. Thanks for addressing my comments.

xen2 commented:
Sorry, don't hesitate to add me as reviewer in case I miss the message next time!

xen2 reviewed (commented):
LGTM!
Maybe I better create a `zig-cross` branch on this repo so you that you can merge the successive PR?

jazzay commented:
> LGTM! Maybe I better create a `zig-cross` branch on this repo so you that you can merge the successive PR?

I think we can just merge straight into master even with these new commits?
